proposals/0132-sequence-end-ops.md

```diff
@@ -16,724 +17,227 @@ these names so they follow consistent, predictable patterns.
-| **Based&nbsp;on&nbsp;Index,&nbsp;Arbitrary**     |
-| (i: Index) ..< (j: Index)                        | Range\<Index>                     | C[i&nbsp;..<&nbsp;j] | C.removeSubrange(i&nbsp;..<&nbsp;j) | C.replaceSubrange(i&nbsp;..<&nbsp;j,&nbsp;with:&nbsp;[x,y]) |
-| &nbsp;&nbsp;...Countable                         | CountableRange\<Index>            | C[i ..< j]           | C.removeSubrange(i&nbsp;..<&nbsp;j) | C.replaceSubrange(i&nbsp;..<&nbsp;j,&nbsp;with:&nbsp;[x,y]) |
-| (i: Index) ... (j: Index)                        | ClosedRange\<Index>               | C[i ... j]           | C.removeSubrange(i ... j)           | C.replaceSubrange(i ... j, with: [x,y])                     |
-| &nbsp;&nbsp;...Countable                         | CountableClosedRange\<Index>      | C[i ... j]           | C.removeSubrange(i ... j)           | C.replaceSubrange(i ... j, with: [x,y])                     |
-| **Based&nbsp;on&nbsp;Index,&nbsp;From&nbsp;End** |
-| startIndex ..< (i: Index)                        | **IncompleteRange\<Index>**       | **C[..\<i]**         | **C.removeSubrange(..\<i)**         | **C.replaceSubrange(..\<i,&nbsp;with:&nbsp;[x,y])**         |
-| (i: Index) ..< endIndex                          | **IncompleteRange\<Index>**       | **C[i..\<]**         | **C.removeSubrange(i..\<)**         | **C.replaceSubrange(i..\<, with: [x,y])**                   |
-| startIndex ... (i: Index)                        | **IncompleteClosedRange\<Index>** | **C[...i]**          | **C.removeSubrange(...i)**          | **C.replaceSubrange(...i, with: [x,y])**                    |
-
-However, it should be implemented with merely:
-
-|                                               | Type                                                | Get                  | Remove                              | Replace                                                     |
-| --------------------------------------------- | --------------------------------------------------- | -------------------- | ----------------------------------- | ----------------------------------------------------------- |
-| (i:&nbsp;Index)&nbsp;..<&nbsp;(j:&nbsp;Index) | Range\<Index>                                       | C[i&nbsp;..<&nbsp;j] | C.removeSubrange(i&nbsp;..<&nbsp;j) | C.replaceSubrange(i&nbsp;..<&nbsp;j,&nbsp;with:&nbsp;[x,y]) |
-| Everything else                               | RangeExpression where&nbsp;Bound&nbsp;==&nbsp;Index | C[i&nbsp;...&nbsp;j] | C.removeSubrange(i&nbsp;...&nbsp;j) | C.replaceSubrange(i&nbsp;...&nbsp;j,&nbsp;with:&nbsp;[x,y]) |
+We should rename these methods to always use `first` or `last` to indicate 
+they operate on a single element, or `prefix` or `suffix` to indicate they 
+operate on many elements. Future APIs should follow this rule as well; for 
+instance, a method which removed and returned *n* leading elements should 
+be called `popPrefix(_:)`, not `popFirst(_:)` or `pop(_:)`.
```

Comment:
I find `popPrefix` to be a very un-compelling example, as there are no known use-cases, but the semantically similar operation that breaks a collection into two slices are a well-known need but should definitely not be spelled the same way.